### PR TITLE
Add LineMessageHandlerSupport to support SpringMVC like event mapping.

### PR DESCRIPTION
**Pull request by kazuki-ma**
Try to add some support class in SDK to easy method dispatching.

`@EnableLineMessaging` make possible to avoid `if (event instanceof XxxEvent) {} else if (event instanceof YyyEvent) {}....` repeating.

Sample
--------
```java
@SpringBootApplication
@LineMessageHandler // ★1: indices this class has event handler method.
public class EchoApplication {
    @Autowired
    private LineMessagingService lineMessagingService;

    public static void main(String[] args) {
        SpringApplication.run(EchoApplication.class, args);
    }

    @EventMapping // ★2: This is handler method for MessageEvent<TextMessageContent>
    public void handleTextMessageEvent(MessageEvent<TextMessageContent> event) throws Exception {
        System.out.println("event: " + event);
        final BotApiResponse apiResponse = lineMessagingService
                .replyMessage(new ReplyMessage(event.getReplyToken(),
                                               singletonList(new TextMessage(event.getMessage().getText()))))
                .execute().body();
        System.out.println("Sent messages: " + apiResponse);
    }

    @EventMapping
    public void handleBeaconEvent(BeaconEvent event) {
        // handle beacon event here.
    }
}
```

### CODE REVIEW

**Comment by tokuhirom** [comment]
* I suggest to remove `@EnableLineMessaging` and use auto configuration.
  * It's spring boot way.
* Could you take event types from method signature instead of annotation parameters?

**Comment by kazuki-ma** [comment]
Updated as a @tokuhirom's suggestion.
* [x]  remove @EnableLineMessaging
* [x] take event types from method signature

File: line-bot-spring-boot/src/main/java/com/linecorp/bot/spring/boot/support/LineMessageHandlerSupport.java [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+    @Value
+    public static class HandlerMethod {
+        Predicate<Event> supportType;
+        Object object;
+        Method handler;
+        int priority;
+    }
+
+    @PostMapping("${line.bot.handler.path:/callback}")
```
Comment (tokuhirom) [inline line-bot-spring-boot/src/main/java/com/linecorp/bot/spring/boot/support/LineMessageHandlerSupport.java]: Okay, I see.
(we were discussed this issue a few days ago, but I forgot to submit the review comment :()

**Comment by be-hase** [comment]
Looks Good.  👍 
This is very convenient feature. 😀 

I think that it is more better to use this feature on not only spring but also other framework(for example, dropwizard, spark,...and so on).

**Comment by kazuki-ma** [comment]
@be-hase Thanks comment.
Extending other framework is interesting but please consider as another topic.

**Comment by tokuhirom** [comment]
👍

**Pull request merged**